kartothek/core/dataset.py

```diff
@@ -180,35 +173,43 @@ def storage_keys(uuid, store):
             )
         )
 
-    def to_dict(self):
+    def to_dict(self) -> Dict:
         dct = OrderedDict(
             [
                 (naming.METADATA_VERSION_KEY, self.metadata_version),
                 (naming.UUID_KEY, self.uuid),
             ]
         )
         if self.indices:
-            dct["indices"] = {k: v.to_dict() for k, v in self.indices.items()}
+            dct["indices"] = {
```

Comment:
what's the reason for changing the logic here?

Reply:
Previously the `to_dict` method returned either the dict or the storage key. That's obviously not type safe and this is where it breaks. To fis this I needed to introduce this complicated dict comprehension